src/utils/get-host-app-url.ts

```diff
@@ -0,0 +1,36 @@
+import http from 'http';
+import url from 'url';
+
+import * as ShopifyErrors from '../error';
+import {Context} from '../context';
+
+/**
+ * Helper method to get the host URL for the app.
+ *
+ * @param request Current HTTP request
+ */
+export default function getHostAppUrl(request: http.IncomingMessage): string {
+  if (!request) {
+    throw new ShopifyErrors.MissingRequiredArgument(
+      'getHostAppUrl requires a request object argument',
+    );
+  }
+
+  if (!request.url) {
+    throw new ShopifyErrors.InvalidRequestError(
+      'Request does not contain a URL',
+    );
+  }
+
+  const query = url.parse(request.url, true).query;
```

Comment:
`url.parse` is showing up as *legacy* though not deprecated (because so many modules rely on it) but it has security concerns: https://nodejs.org/api/url.html#urlparseurlstring-parsequerystring-slashesdenotehost

Wondering if this should be 
```suggestion
  const host = new URL(request.url).searchParams.get('host');
```
(which is terribly clunky 😞)

Reply:
Thanks for spotting that.  Ive updated this to use `URL`.

Reply:
Is it possible that `request.url` will not be absolute?  If it will always be absolute, then the first line could be simplified to

```typescript
const url = new URL(request.url);
```

Otherwise, this seems good to me!

Reply:
If we need to keep the `base` parameter for the `new URL()` call, should it be `https` instead of `http`?

Reply:
In my testing the express template passed a relative URL.

I'll update it to use https though.  I don't think it would matter as we don't use that portion of the URL, but https is a better default.